### PR TITLE
feat: custom metrics for check results

### PR DESCRIPTION
**Pull request by yashmehrotra**

### CODE REVIEW

File: checks/metrics.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+			logger.Errorf("Invalid type for check.metrics %s for check[%s]", metricsSpec.Type, c.GetName())
+			return
+		}
+	}
+
+	tplValue := v1.Template{Template: metricsSpec.Value}
+	for _, r := range results {
+		templateInput := map[string]any{
+			"result": r.Data,
+			"check": map[string]any{
```
Comment (moshloop) [inline checks/metrics.go:86]: Probably need a duration in here as well

File: fixtures/minimal/http_pass_single.yaml [hunk 2]
```diff
@@ -11,6 +11,14 @@ spec:
       name: http-deprecated-endpoint
     - name: http-minimal-check
       url: https://httpbin.demo.aws.flanksource.com/status/200
+      metrics:
```
Comment (moshloop) [inline fixtures/minimal/http_pass_single.yaml:14]: can we also add examples for junit e.g. number of tests, passing, failing, time

File: checks/metrics.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+				"name":        c.GetName(),
+				"description": c.GetDescription(),
+				"labels":      c.GetLabels(),
+				"endpoint":    c.GetEndpoint(),
+			},
+		}
+		valRaw, err := template(ctx.New(templateInput), tplValue)
+		if err != nil {
+			logger.Errorf("Error templating value for check.metrics template %s for check[%s]", metricsSpec.Value, c.GetName())
```
Comment (adityathebe) [inline checks/metrics.go:95]: need to log the error as well. And should we proceed with execution after an error? I think we need a continue

File: fixtures/minimal/http_pass_single.yaml (hunk 2 above)
Comment (yashmehrotra) [reply fixtures/minimal/http_pass_single.yaml:14]: Added

File: api/external/metrics.go [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package external
+
+// +kubebuilder:object:generate=true
+type Metrics struct {
```
[the remark this answers is not in the document]
Comment (yashmehrotra) [reply api/external/metrics.go:4]: Fixed

File: checks/metrics.go (hunk 1 above)
Comment (yashmehrotra) [reply checks/metrics.go]: Added

**Pull request merged**